tensorflow/core/nccl/nccl_manager.cc

```diff
@@ -178,9 +178,13 @@ struct NcclManager::Collective : public core::RefCounted {
     // or to manage one non-singleton NcclManager instance.
     // For example, the nccl_manager_test will use both paradigms in the same
     // executable, but not running concurrently (which would hang otherwise).
+    // Temporarily backing out the assert in favor of a warning.
+    // TODO: reinstate the assert with a more targeted conditional.
     if (NcclManager::instance_count > 1) {
-      status = errors::Internal(
-          "ROCm cannot use multi-node NCCL collectives on a single node");
+      static absl::once_flag once;
+      absl::call_once(once, [] {
+        LOG(WARNING) << "ROCm cannot use multi-node NCCL collectives on a single node";
```

Comment:
we need to change the wording of the warning message..something like

```
More than one concurrent instance of NcclManager has been instantiated on the same node, this can sometimes result in a race condition leading to a system hang.
```

@jeffdaily what say you?